packages/vue3/src/components/Tooltip/Tooltip.vue

```diff
@@ -1,13 +1,73 @@
 <script setup lang="ts">
-defineProps<{
+import { onMounted, onUnmounted, ref } from 'vue';
+
+const props = withDefaults(defineProps<{
   label?: string
   position?: 'left' | 'right' | 'top' | 'bottom'
-}>();
+}>(), {
+  position: 'top',
+});
+const tooltipTrigger = ref<HTMLElement>();
+const tooltip = ref<HTMLElement>();
+
+const setTooltipPosition = (position = props.position) => {
```

Comment:
This function seems to do more than one thing, it would be better to dissect it into smaller functions

Reply:
Maybe colocate a utils.ts file where you keep those functions 🤔